wolfCLU/src/x509/clu_request_setup.c

```diff
@@ -0,0 +1,292 @@
+                    ret = -1;
+                }
+                break;
+
+            case WOLFCLU_INFORM:
+                inForm = wolfCLU_checkInform(optarg);
+                (void)inForm; /* for future use */
+                break;
+
+            case WOLFCLU_OUTFORM:
+                outForm = wolfCLU_checkOutform(optarg);
+                break;
+
+            case WOLFCLU_HELP:
+                wolfCLU_certgenHelp();
+                return 0;
+
+            case WOLFCLU_RSA:
+                algCheck = 1;
+                break;
+
+            case WOLFCLU_ECC:
+                algCheck = 3;
+                break;
+
+            case WOLFCLU_ED25519:
+                algCheck = 2;
+                break;
+
+            case WOLFCLU_CONFIG:
+                config = optarg;
+                break;
+
+            case WOLFCLU_DAYS:
+                days = atoi(optarg);
+                break;
+
+            case WOLFCLU_CERT_SHA:
+                md  = wolfSSL_EVP_sha1();
+                oid = SHA_HASH;
+                break;
+
+            case WOLFCLU_CERT_SHA224:
+                md  = wolfSSL_EVP_sha224();
+                oid = SHA_HASH224;
+                break;
+
+            case WOLFCLU_CERT_SHA256:
+                md  = wolfSSL_EVP_sha256();
+                oid = SHA_HASH256;
+                break;
+
+            case WOLFCLU_CERT_SHA384:
+                md  = wolfSSL_EVP_sha384();
+                oid = SHA_HASH384;
+                break;
+
+            case WOLFCLU_CERT_SHA512:
+                md  = wolfSSL_EVP_sha512();
+                oid = SHA_HASH512;
+                break;
+
+            case WOLFCLU_X509:
+                genX509 = 1;
+                break;
+
+            case ':':
+            case '?':
+                break;
+
+            default:
+                /* do nothing. */
+                (void)ret;
+        }
+    }
+
+    /* default to sha256 if not set */
+    if (md == NULL) {
+        md  = wolfSSL_EVP_sha256();
+        oid = SHA_HASH256;
+    }
+
+    x509 = wolfSSL_X509_new();
+    if (x509 == NULL) {
+        printf("issue creating structure to use\n");
+        ret = MEMORY_E;
+    }
+
+    if (ret == 0 && days > 0) {
+        WOLFSSL_ASN1_TIME *notBefore, *notAfter;
+        time_t t;
+
+        t = time(NULL);
+        notBefore = wolfSSL_ASN1_TIME_adj(NULL, t, 0, 0);
+        notAfter = wolfSSL_ASN1_TIME_adj(NULL, t, atoi(argv[ret+1]), 0);
+        if (notBefore == NULL || notAfter == NULL) {
+            printf("error creating not before/after dates\n");
+            ret = -1;
+        }
+        else {
+            wolfSSL_X509_set_notBefore(x509, notBefore);
+            wolfSSL_X509_set_notAfter(x509, notAfter);
+        }
+
+        wolfSSL_ASN1_TIME_free(notBefore);
+        wolfSSL_ASN1_TIME_free(notAfter);
+    }
+
+
+    if (ret == 0 && bioIn != NULL) {
+        pkey = wolfSSL_PEM_read_bio_PrivateKey(bioIn, NULL, NULL, NULL);
```

Comment:
Why read a in private key mode if you use it to set a public key?

Reply:
In the current use case the certificate generated is a self signed cert. With req -new -x509.